src/Adapters/Etw.Net451/EtwTelemetryModule.cs

```diff
@@ -0,0 +1,246 @@
+    using Microsoft.Diagnostics.Tracing;
+    using Microsoft.Diagnostics.Tracing.Session;
+
+    /// <summary>
+    /// A module to trace data submitted via .NET framework <seealso cref="Microsoft.Diagnostics.Tracing.Session" /> class.
+    /// </summary>
+    public class EtwTelemetryModule : ITelemetryModule, IDisposable
+    {
+        private readonly object lockObject;
+        private TelemetryClient client;
+        private bool isDisposed = false;
+        private bool isInitialized = false;
+        private ITraceEventSession traceEventSession;
+        private List<Guid> enabledProviderIds;
+        private List<string> enabledProviderNames;
+
+        /// <summary>
+        /// EtwTelemetryModule default constructor.
+        /// </summary>
+        public EtwTelemetryModule() : this(
+            new AITraceEventSession(new TraceEventSession(string.Format(CultureInfo.InvariantCulture, "ApplicationInsights-{0}-{1}", nameof(EtwTelemetryModule), Guid.NewGuid()))),
+            new Action<ITraceEventSession, TelemetryClient>((traceSession, client) =>
+            {
+                if (traceSession != null && traceSession.Source != null)
+                {
+                    traceSession.Source.Dynamic.All += traceEvent =>
+                    {
+                        traceEvent.Track(client);
+                    };
+                    traceSession.Source.Process();
+                }
+            }))
+        {
+        }
+
+        internal EtwTelemetryModule(
+            ITraceEventSession traceEventSession,
+            Action<ITraceEventSession, TelemetryClient> startTraceEventSessionAction)
+        {
+            this.lockObject = new object();
+            this.Sources = new List<EtwListeningRequest>();
+            this.enabledProviderIds = new List<Guid>();
+            this.enabledProviderNames = new List<string>();
+
+            this.traceEventSession = traceEventSession;
+            this.StartTraceEventSession = startTraceEventSessionAction;
+        }
+
+        /// <summary>
+        /// Gets the list of ETW Provider listening requests (information about which providers should be traced).
+        /// </summary>
+        public IList<EtwListeningRequest> Sources { get; private set; }
+
+        private Action<ITraceEventSession, TelemetryClient> StartTraceEventSession
+        {
+            get;
+            set;
+        }
+
+        /// <summary>
+        /// Initializes the telemetry module and starts tracing ETW events specified via <see cref="Sources"/> property.
+        /// </summary>
+        /// <param name="configuration">Module configuration.</param>
+        public void Initialize(TelemetryConfiguration configuration)
+        {
+            string errorMessage;
+            if (configuration == null)
+            {
+                errorMessage = string.Format(CultureInfo.InvariantCulture, "Argument {0} is required. The initialization is terminated.", nameof(configuration));
```

Comment:
Ahah, I know someone is going to ask this.
Firstly, output of ToString() various on cultures that I can't use string interpolation directly. The intended way to use string interpolation with IFormatter is to make use of System.FormattableString, which, unfortunately, does not exist in .NET Fx 4.5.1.

Unless you have a better way to go, I'll have to keep using string.Format here. :-)